// chromadb/dist/main/generated/models.js
"use strict";
/* eslint-disable */
// tslint:disable
/**
 * FastAPI
 *
 *
 * OpenAPI spec version: 0.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator+.
 * https://github.com/karlvr/openapi-generator-plus
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Api = void 0;
var Api;
(function (Api) {
    /**
     * @export
     * @namespace GetEmbedding
     */
    let GetEmbedding;
    (function (GetEmbedding) {
        /**
         * @export
         * @namespace Include
         */
        let Include;
        (function (Include) {
            let EnumValueEnum;
            (function (EnumValueEnum) {
                EnumValueEnum["Documents"] = "documents";
            })(EnumValueEnum = Include.EnumValueEnum || (Include.EnumValueEnum = {}));
            let EnumValueEnum2;
            (function (EnumValueEnum2) {
                EnumValueEnum2["Embeddings"] = "embeddings";
            })(EnumValueEnum2 = Include.EnumValueEnum2 || (Include.EnumValueEnum2 = {}));
            let EnumValueEnum3;
            (function (EnumValueEnum3) {
                EnumValueEnum3["Metadatas"] = "metadatas";
            })(EnumValueEnum3 = Include.EnumValueEnum3 || (Include.EnumValueEnum3 = {}));
            let EnumValueEnum4;
            (function (EnumValueEnum4) {
                EnumValueEnum4["Distances"] = "distances";
            })(EnumValueEnum4 = Include.EnumValueEnum4 || (Include.EnumValueEnum4 = {}));
        })(Include = GetEmbedding.Include || (GetEmbedding.Include = {}));
    })(GetEmbedding = Api.GetEmbedding || (Api.GetEmbedding = {}));
    /**
     * @export
     * @namespace QueryEmbedding
     */
    let QueryEmbedding;
    (function (QueryEmbedding) {
        /**
         * @export
         * @namespace Include
         */
        let Include;
        (function (Include) {
            let EnumValueEnum;
            (function (EnumValueEnum) {
                EnumValueEnum["Documents"] = "documents";
            })(EnumValueEnum = Include.EnumValueEnum || (Include.EnumValueEnum = {}));
            let EnumValueEnum2;
            (function (EnumValueEnum2) {
                EnumValueEnum2["Embeddings"] = "embeddings";
            })(EnumValueEnum2 = Include.EnumValueEnum2 || (Include.EnumValueEnum2 = {}));
            let EnumValueEnum3;
            (function (EnumValueEnum3) {
                EnumValueEnum3["Metadatas"] = "metadatas";
            })(EnumValueEnum3 = Include.EnumValueEnum3 || (Include.EnumValueEnum3 = {}));
            let EnumValueEnum4;
            (function (EnumValueEnum4) {
                EnumValueEnum4["Distances"] = "distances";
            })(EnumValueEnum4 = Include.EnumValueEnum4 || (Include.EnumValueEnum4 = {}));
        })(Include = QueryEmbedding.Include || (QueryEmbedding.Include = {}));
    })(QueryEmbedding = Api.QueryEmbedding || (Api.QueryEmbedding = {}));
})(Api || (exports.Api = Api = {}));
